chapter-outline.md

```diff
@@ -0,0 +1,287 @@
+        in our courses in general in MDS,
+        so this would be a great opportunity to try out the concept.
+
+## Thoughts and questions
+
+- We want chapters to have a familiar structure to students
+  so that they know what to expect as they progress in the book,
+  but we don't want them to get bored.
+    - One thing I wonder is if there is value in that the chapter design evolves over time
+      akin to how levels in video games don't follow quite the same pattern
+      even if there are some similar elements like boss fights etc.
+      Is there value in introducing this novelty to make students stay engaged?
+      I think learning apps like Duolingo also have slightly different level design
+      although not as extensive as in a good video game.
+        - I'm curious about this,
+          but it is an "out there" idea I would say
+          and we would be on uncharted waters
+          if we start to evolve the book structure
+          as the story progresses.
+          I think it sounds fun,
+          and someone must have written something about this idea in the past
+          so maybe we can find a few academic articles to learn more
+          if it is considered an effective learning strategy.
+            - Maybe we can spend a little time looking at the idea,
+              but I don't want this to be a time sink up front
+              (although the more I think about it the more it enthralls me =p).
+- As part of the grant proposal I wrote about creating videos to enhance students learning.
+    - I don't think we are strictly bound by this,
+      but if we can think of ways where this complements what students can learn from the text
+      then I would like to include it
+      as I think it can also mix up the content more
+      and engage via different modalities.
+        - My initial idea for these videos is around me drawing charts and concepts.
+          I think the process of drawing
+          is something that students cannot get from a static resource,
+          since that will just show the output.
+          It could shed light on some conceptual aspect
+          and allows students to see aspect laid out more visually.
+          For example,
+          I drew out a schematic for how all git commands are connecting
+          when I originally taught 521,
+          and it was really helpful for students to get a graphical overview of what is happening.
+- Should we enable comments on the entire book? There are technologies such as `hypothes.is`
+  that are easily enabled in quarto
+  and allows student to highlight sections in the book and take notes.
+  They can even opt to make their notes public to other students
+  and have discussions in the comments
+  (maybe preferential and more direct vs having discussion on slack?),
+  but there would be no notifications and we are splitting discussion into two avenues.
+  Maybe more a space to exchange reflections on the content?
```

Comment:
I like the idea of students being able to share reflections. hypothes could be good if it's unobtrusive. probably only very keen students would be interested in this.

I was also wondering if students can sort of annotate the charts in the book when identifying problems etc. I don't know how this would work technologically.

Reply:
Needs critical mass to be interesting. Unlikely is not mandatory? Useful outside MDS.

Unsure how it works when the content changes...